### ISSUE TITLE
Pass url from partial

### ISSUE THREAD

**Issue by zackify** [Legitcode/legible#21]
Say you have a bunch of requests that you define like this:

```js
let test = partial`
    url: /auth/login
    method: POST
    body: ${{ email, password }}
  `
```
relative paths.

It would be awesome to do: 

```js
let response = await test`
      url: ${url => `${process.env.API}/${url}`}
      headers: ${{
        Authorization: 'Bearer: token'
      }}
    `
```
This way you can prefix urls and get the url from the partial.

Here's the full example of what I'm doing in case you're curious
```js
import base from 'requests/base'
import { partial } from 'legible'

export async function login({ email, password }) {
  let request = partial`
    url: /auth/login
    method: POST
    body: ${{ email, password }}
  `

  return await base(request)
}

//requests/base.js

export default async function request(partial) {
  try {
    let response = await partial`
      url: ${url => `${process.env.API}/${url}`}
      headers: ${{
        Authorization: 'Bearer: token'
      }}
    `
    return { response }
  }
  catch(error) {
    console.log(error)
    return { error }
  }
}

```

This way I don't need to try catch everywhere and I can check for `error` on all requests. @raygesualdo

**Comment by raygesualdo** [comment]
Essentially, you want a way to define a base URL? Am I understanding that correctly?

**Comment by zackify** [comment]
Yes, i want to be able to pass a function for the url to get the original url from a partial. I made a pr and merged it for now. We can make it so any option can have a callback like this. could be useful.

**Comment by zackify** [comment]
closed by #23
